test/test_sync.py

```diff
@@ -107,10 +107,15 @@ def test_broken_symlink(self):
     def test_invalid_permissions(self):
         with TempDir() as tmpdir:
             folder = self._prepare_folder(tmpdir, invalid_permissions=True)
-            self.assertEqual(self.NAMES[1:], list(f.name for f in folder.all_files(self.reporter)))
-            self.reporter.local_permission_error.assert_called_once_with(
-                os.path.join(tmpdir, self.NAMES[0])
-            )
+            if not os.access(os.path.join(tmpdir, self.NAMES[0]), os.R_OK):
```

Comment:
A comment on this test would be good.  If I come back to this code a year from now, I don't think I'll remember what's going on here.